Build error on release-v0.3_0.18 branch due to C++ 17

I'm getting this error building on the release-v0.3_0.18 branch:

```
In file included from /Users/woodser/git/monero-lws/src/db/string.cpp:30:
/Users/woodser/git/monero-java/external/monero-cpp/external/monero-project/src/cryptonote_basic/cryptonote_basic_impl.h:43:29: error: no template named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
  struct array_hasher: std::unary_function<t_array&, std::size_t>
                       ~~~~~^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__functional/unary_function.h:46:1: note: '__unary_function' declared here
using __unary_function = __unary_function_keep_layout_base<_Arg, _Result>;
^
```

The error resolves by changing `set(CMAKE_CXX_STANDARD 17)` to `set(CMAKE_CXX_STANDARD 14)`.

Guessing it's too far ahead of monero-project?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ cmake_minimum_required(VERSION 3.1.0)
 project(monero-lws)
 
 enable_language(CXX)
-set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 option(BUILD_TESTS "Build Tests" OFF)

--- a/tests/unit/rest.test.cpp
+++ b/tests/unit/rest.test.cpp
@@ -27,7 +27,7 @@
 
 #include "framework.test.h"
 
-#include <optional>
+#include <boost/optional/optional.hpp>
 #include "db/data.h"
 #include "db/storage.test.h"
 #include "db/string.h"
@@ -111,7 +111,7 @@ LWS_CASE("rest_server")
 
   SETUP("Database and login")
   {
-    std::optional<lws::rest_server> server;
+    boost::optional<lws::rest_server> server;
     lws::db::test::cleanup_db on_scope_exit{};
     lws::db::storage db = lws::db::test::get_fresh_db();
     auto context =

--- a/tests/unit/wire/read.test.cpp
+++ b/tests/unit/wire/read.test.cpp
@@ -47,7 +47,7 @@ namespace
     {
       EXPECT(Target(0) == wire::integer::cast_unsigned<Target>(std::uintmax_t(0)));
       EXPECT(limit::max() == wire::integer::cast_unsigned<Target>(std::uintmax_t(limit::max())));
-      if constexpr (limit::max() < max)
+      if (limit::max() < max)
       {
         EXPECT_THROWS_AS(wire::integer::cast_unsigned<Target>(std::uintmax_t(limit::max()) + 1), wire::exception);
         EXPECT_THROWS_AS(wire::integer::cast_unsigned<Target>(max), wire::exception);
@@ -68,7 +68,7 @@ namespace
 
     SETUP("intmax_t to " + boost::core::demangle(typeid(Target).name()))
     {
-      if constexpr (min < limit::min())
+      if (min < limit::min())
       {
         EXPECT_THROWS_AS(wire::integer::cast_signed<Target>(std::intmax_t(limit::min()) - 1), wire::exception);
         EXPECT_THROWS_AS(wire::integer::cast_signed<Target>(min), wire::exception);
@@ -76,7 +76,7 @@ namespace
       EXPECT(limit::min() == wire::integer::cast_signed<Target>(std::intmax_t(limit::min())));
       EXPECT(Target(0) == wire::integer::cast_signed<Target>(std::intmax_t(0)));
       EXPECT(limit::max() == wire::integer::cast_signed<Target>(std::intmax_t(limit::max())));
-      if constexpr (limit::max() < max)
+      if (limit::max() < max)
       {
         EXPECT_THROWS_AS(wire::integer::cast_signed<Target>(std::intmax_t(limit::max()) + 1), wire::exception);
         EXPECT_THROWS_AS(wire::integer::cast_signed<Target>(max), wire::exception);

